### PR TITLE
update particles

### PR DESCRIPTION
**Pull request by songchengjiang**

### CODE REVIEW

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ void PUNoise3D::initialise(unsigned short octaves, double frequency, double ampl
 //	double y = 0.0;
 //	double step = 1.0 / (double)dimension;
 //	size_t buffSize = 4 * dimension * dimension; // Assume Image of pixelformat 32 bits (i.e. PF_R8G8B8A8)
-//	unsigned char* buff = new (std::nothrow) unsigned char[buffSize];
+//	unsigned char* buff = new unsigned char[buffSize];
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp]: nothrow

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp [hunk 2]
```diff
@@ -93,7 +93,7 @@ void PUNoise3D::initialise(unsigned short octaves, double frequency, double ampl
 //			y += step;
 //		}
 //	}
-//	Ogre::Image* image = new (std::nothrow) Ogre::Image();
+//	Ogre::Image* image = new Ogre::Image();
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp]: nothrow

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptCompiler.cpp [hunk 3]
```diff
@@ -162,7 +162,7 @@ PUAtomAbstractNode::PUAtomAbstractNode(PUAbstractNode *ptr)
 
 PUAbstractNode *PUAtomAbstractNode::clone() const
 {
-    PUAtomAbstractNode *node = new (std::nothrow) PUAtomAbstractNode(parent);
+    PUAtomAbstractNode *node = new PUAtomAbstractNode(parent);
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptCompiler.cpp]: nothrow

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp [hunk 4]
```diff
@@ -57,6 +56,8 @@ void traceScriptParser(PUConcreteNodeList& nodes)
     
 }
 
+
+//解析脚本,生成nodeList//
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: remove chinese

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp [hunk 5]
```diff
@@ -155,6 +156,10 @@ void PUScriptParser::parse(PUConcreteNodeList& nodes,const PUScriptTokenList& to
                         if(i == end || (*i)->type != TID_VARIABLE)
                         {
                             printf("Exception");
+//							OGRE_EXCEPT(Exception::ERR_INVALID_STATE,
+//                                        Ogre::String("expected variable name at line ") +
+//                                        Ogre::StringConverter::toString(node->line),
+//                                        "ScriptParser::parse");
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: unused code

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp [hunk 6]
```diff
@@ -507,6 +512,10 @@ void PUScriptParser::parseChunk(PUConcreteNodeList& nodes, const PUScriptTokenLi
                 node->type = CNT_QUOTE;
             default:
                 printf("unexpected token,%s,%d\n",token->lexeme.c_str(),token->line);
+//				OGRE_EXCEPT(Exception::ERR_INVALID_STATE, 
+//                            Ogre::String("unexpected token") + token->lexeme + " at line " + 
+//                            Ogre::StringConverter::toString(token->line),
+//                            "ScriptParser::parseChunk");
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: unused code

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptTranslator.cpp [hunk 7]
```diff
@@ -347,7 +346,7 @@ bool PUScriptTranslator::passValidatePropertyNumberOfValuesRange(PUScriptCompile
 //        compiler->addError(ScriptCompiler::CE_FEWERPARAMETERSEXPECTED,
 //                           prop->file,
 //                           prop->line,
-//                           "PU Compiler: " + token + " must have between" +
+//                           "PU Compiler: " + token2 + " must have between" +
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptTranslator.cpp]: remove it or use token2
parameter here is token2

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DTranslateManager.h [hunk 8]
```diff
@@ -27,15 +27,18 @@
 #define __CC_PU_PARTICLE_3D_TRANSLATE_MANAGER_H__
 
 #include "base/CCRef.h"
-//#include "extensions/Particle3D/ParticleUniverse/Script/Translater/CCParticle3DSystemTranslator.h"
-//#include "extensions/Particle3D/ParticleUniverse/Script/Translater/CCAliasTranslator.h"
-//#include "extensions/Particle3D/ParticleUniverse/Script/Translater/CCParticle3DTechniqueTranslator.h"
+//#include "3dparticle/ParticleUniverse/Script/Translater/CCParticle3DSystemTranslator.h"
+//#include "3dparticle/ParticleUniverse/Script/Translater/CCAliasTranslator.h"
+//#include "3dparticle/ParticleUniverse/Script/Translater/CCParticle3DTechniqueTranslator.h"
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DTranslateManager.h]: correct this

File: extensions/Particle3D/ParticleUniverse/CCPUParticleSystem3D.cpp [hunk 9]
```diff
@@ -198,72 +285,47 @@ PUParticleSystem3D* PUParticleSystem3D::create( const std::string &filePath )
 void PUParticleSystem3D::startParticleSystem()
 {
     stopParticleSystem();
-    _particlePool.lockAllParticles();
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/CCPUParticleSystem3D.cpp:201]: should keep lockAllParticles();

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp (hunk 1 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp (hunk 2 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DNoise.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptCompiler.cpp (hunk 3 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptCompiler.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp (hunk 4 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp (hunk 5 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp (hunk 6 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptParser.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptTranslator.cpp (hunk 7 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DScriptTranslator.cpp]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticle3DTranslateManager.h (hunk 8 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticle3DTranslateManager.h]: fixed

File: extensions/Particle3D/ParticleUniverse/CCPUParticleSystem3D.cpp (hunk 9 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/CCPUParticleSystem3D.cpp:201]: use forceStopParticleSystem() instead

File: extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.cpp [hunk 10]
```diff
@@ -389,12 +428,7 @@ void PUParticle3DModelRender::render( Renderer* renderer, const Mat4 &transform,
 
     if (_spriteList.empty()){
         for (unsigned int i = 0; i < particleSystem->getParticleQuota(); ++i){
-            auto sprite = Sprite3D::create(_modelFile);
-            if (sprite == nullptr)
-            {
-                CCLOG("failed to load file %s", _modelFile.c_str());
-                continue;
-            }
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.cpp:397]: check

File: extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.h [hunk 11]
```diff
@@ -97,10 +161,11 @@ class CC_DLL PUParticle3DQuadRender : public Particle3DQuadRender
 
     void getOriginOffset(int &offsetX, int &offsetY);
     void determineUVCoords(PUParticle3D *particle);
-    void fillVertex(unsigned short index, const Vec3 &pos, const Vec4 &color, const Vec2 &uv);
+    void fillVertex(unsigned short index, const cocos2d::Vec3 &pos, const cocos2d::Vec4 &color, const cocos2d::Vec2 &uv);
```
Comment (super626) [inline extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.h]: already in cocos2d namespace

File: extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.cpp (hunk 10 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.cpp:397]: fixed

File: extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.h (hunk 11 above)
Comment (songchengjiang) [inline extensions/Particle3D/ParticleUniverse/ParticleRenders/CCPUParticle3DRender.h]: fixed

**Pull request merged**